src/provider/handler/getPosts.js

```diff
@@ -1,18 +1,34 @@
 import axios from 'axios';
 import {mapPost} from './mappers/postMapper';
+import _url from 'url';
 
 export function getPosts(params) {
-  const {url, categories} = params;  
+  const {url} = params;  
   return new Promise((resolve, reject) => {
```

Comment:
you don't really need the promise constructor here. You can run the code (if you want to reject you can do `return Promise.reject({...});`

just parse your url here, and when it's formed do `return axios.get('...').then()...`